tests/bluetooth/mesh/sbr/src/main.c

```diff
@@ -0,0 +1,409 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/net/buf.h>
+#include <zephyr/bluetooth/mesh.h>
+#include <stdlib.h>
+
+#include "settings.h"
+#include "brg_cfg.h"
+
+/* Used for cleaning RPL without checking it. */
+static bool skip_delete;
+
+#define TEST_VECT_SZ (CONFIG_BT_MESH_BRG_TABLE_ITEMS_MAX + 1)
+
+static struct test_brg_cfg_row {
+	uint8_t direction;
+	uint16_t netidx1;
+	uint16_t netidx2;
+	uint16_t addr1;
+	uint16_t addr2;
+} test_vector[TEST_VECT_SZ];
+
+#define ADDR1_BASE (1)
+#define ADDR2_BASE (100)
+
+/**** Helper functions ****/
+static void setup(void *f)
+{
+	/* create test vector */
+	for (int i = 0; i < TEST_VECT_SZ; i++) {
+		test_vector[i].direction = i < (TEST_VECT_SZ / 2) ? 1 : 2;
+		test_vector[i].netidx1 = (i/8);
+		test_vector[i].addr1 = ADDR1_BASE + i;
+		test_vector[i].netidx2 = (i/8) + 16;
+		test_vector[i].addr2 = ADDR2_BASE + i;
+	}
+
+}
+
+/**** Mocked functions ****/
+
+void bt_mesh_settings_store_schedule(enum bt_mesh_settings_flag flag)
+{
+	ztest_check_expected_value(flag);
+}
+
+int settings_save_one(const char *name, const void *value, size_t val_len)
+{
+	ztest_check_expected_data(name, strlen(name));
+	ztest_check_expected_value(val_len);
+	ztest_check_expected_data(value, val_len);
+	return 0;
+}
+
+int settings_delete(const char *name)
+{
+	if (skip_delete) {
+		return 0;
+	}
+
+	ztest_check_expected_data(name, strlen(name));
+	return 0;
+}
+
+static void check_fill_all_bt_entries(void)
+{
+	int err;
+
+	for (int i = 0; i < TEST_VECT_SZ; i++) {
+
+		if (i < CONFIG_BT_MESH_BRG_TABLE_ITEMS_MAX) {
+			ztest_expect_value(bt_mesh_settings_store_schedule, flag,
+				BT_MESH_SETTINGS_SBR_PENDING);
+		}
+
+		err = brg_cfg_tbl_add(test_vector[i].direction, test_vector[i].netidx1,
+			test_vector[i].netidx2, test_vector[i].addr1, test_vector[i].addr2);
+
+		if (i != CONFIG_BT_MESH_BRG_TABLE_ITEMS_MAX) {
```

Comment:
Yes. `i` will be larger at the end of the loop, because TEST_VECT_SZ is defined as `CONFIG_BT_MESH_BRG_TABLE_ITEMS_MAX + 1`, so that we can test that adding extra entries actually fails.